File: problems/0011_Container_With_Most_Water/xiaoheng_3.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
	
	int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int maxarea = 0;
        
        while(left != right){
            maxarea = max(min(height[left],height[right]) * (right-left) , maxarea);
            
            (height[left] < height[right]) ? ++left : --right;
            
        }
        return maxarea;
	
	return 0;
} 

File: problems/0717_1-bit and 2-bit Characters/xiaoheng_3.js
var isOneBitCharacter = function(bits) {
    let i = 0
    
    while(i < bits.length-1){
        i = bits[i] === 0 ? i + 1 : i + 2
    }
    
    return i === bits.length-1 && bits[bits.length-1] === 0
};